.github/workflows/unittests.yml

```diff
@@ -1,75 +0,0 @@
-# This workflow is used to run the unittest of pyiron
-
-name: Unittests
-
-on:
-  push:
-    branches: [ main ]
-  pull_request:
-    branches: [ main ]
-
-jobs:
-  build:
-    runs-on: ${{ matrix.operating-system }}
-    strategy:
-      matrix:
-        include:
-        - operating-system: macos-latest
-          python-version: '3.12'
-          label: osx-64-py-3-12
-          prefix: /Users/runner/miniconda3/envs/my-env
-
-        - operating-system: windows-latest
-          python-version: '3.12'
-          label: win-64-py-3-12
-          prefix: C:\Miniconda3\envs\my-env
-
-        - operating-system: ubuntu-latest
-          python-version: '3.12'
-          label: linux-64-py-3-12
-          prefix: /usr/share/miniconda3/envs/my-env
-          
-        - operating-system: ubuntu-latest
-          python-version: '3.11'
-          label: linux-64-py-3-11
-          prefix: /usr/share/miniconda3/envs/my-env
-          
-        - operating-system: ubuntu-latest
-          python-version: '3.10'
-          label: linux-64-py-3-10
-          prefix: /usr/share/miniconda3/envs/my-env
-          
-        - operating-system: ubuntu-latest
-          python-version: 3.9
-          label: linux-64-py-3-9
-          prefix: /usr/share/miniconda3/envs/my-env
-
-    steps:
-    - uses: actions/checkout@v4
-    - name: Setup Mambaforge
-      uses: conda-incubator/setup-miniconda@v2
-      with:
-        python-version: ${{ matrix.python-version }}
-        miniforge-variant: Mambaforge
-        channels: conda-forge
-        channel-priority: strict
-        activate-environment: my-env
-        environment-file: .ci_support/environment.yml
-        use-mamba: true
-    - name: Test
-      shell: bash -l {0}
-      timeout-minutes: 5
```

Comment:
I am missing the option to set the `timeout-minutes` in the new actions.

Reply:
This is a bit trickier. [Github currently does not support passing `timeout-minutes` to composite workflows](https://github.com/orgs/community/discussions/50481). What we _can_ do is leverage it in `pyiron/actions/unit-tests` in the same step you do here, then expose it as an input variable. For the workhorse `pyiron/.github/workflows/push-pull.yml` I wouldn't want an explosion of input options, but am comfortable with a single timeout that gets applied to all invocations of the unit tests.